src/impl/configurator_from_yaml.cpp

```diff
@@ -409,6 +410,25 @@ namespace soralog {
       }
     }
 
+    auto at_fault_node = sink_node["at_fault"];
+    if (at_fault_node.IsDefined()) {
+      if (not at_fault_node.IsScalar()) {
+        errors_ << "W: Property 'at_fault' of sink node is not scalar\n";
+        has_warning_ = true;
+      } else {
+        auto at_fault_str = at_fault_node.as<std::string>();
+        if (at_fault_str == "terminate") {
+          at_fault = Sink::AtFaultReactionType::TERMINATE;
+        } else if (at_fault_str == "ignore") {
+          at_fault = Sink::AtFaultReactionType::DROP_BUFFER;
+        } else if (at_fault_str != "wait") {
```

Comment:
I think we had a function somewhere for this type of CLI param checking. Written by hand I believe it's pretty prone to errors when new values are added.

Reply:
It's not CLI options, just config yaml-file

Reply:
Well, any param checking.